tests/transports/stream_tests.nim

```diff
@@ -200,6 +200,61 @@ template streamTransportTest*(
       ma, transportProvider, streamProvider, serverStreamHandler, clientStreamHandler
     )
 
+  asyncTest "closing session should close all streams":
+    let ma = @[MultiAddress.init(address).tryGet()]
+    const numStreams = 20
+    const numIncomplete = 10
+    const numComplete = numStreams - numIncomplete
+    var successfulReadsFut = newFuture[void]()
+    var successfulReadsCount = 0
+
+    proc serverStreamHandler(stream: Connection) {.async: (raises: []).} =
+      noExceptionWithStreamClose(stream):
+        try:
+          var buffer: array[clientMessage.len, byte]
+          await stream.readExactly(addr buffer[0], clientMessage.len)
+          check string.fromBytes(buffer) == clientMessage
+
+          successfulReadsCount += 1
+          if successfulReadsCount == numComplete:
+            successfulReadsFut.complete()
+        except LPStreamIncompleteError:
+          discard
+
+    proc runClient(server: Transport) {.async.} =
+      let client = transportProvider()
+      let conn = await client.dial(server.addrs[0])
+      let muxer = streamProvider(client, conn)
+      let muxerTask = muxer.handle()
+
+      # 10 streams: send incomplete messages (will block)
+      for i in 0 ..< numIncomplete:
+        let stream = await muxer.newStream()
+        await stream.write(clientMessage[0 ..< 10])
```

Comment:
would be nicer to name `clientMessage[0 ..< 10]`  as `clientIncompleteMessage` const; because this `10` might be confusing to reader, especially with comment 3 lines above.